mesonbuild/optinterpreter.py

```diff
@@ -84,9 +84,27 @@ def ComboParser(name, description, kwargs):
             raise OptionException('Combo choice elements must be strings.')
     return coredata.UserComboOption(name, description, choices, kwargs.get('value', choices[0]))
 
+@permitted_kwargs({'value', 'choices'})
+def string_array_parser(name, description, kwargs):
+    if 'choices' not in kwargs:
+        raise OptionException('Array option missing "choices" keyword.')
+    choices = kwargs['choices']
+    if not isinstance(choices, list):
+        raise OptionException('Array choices must be an array.')
+    for i in choices:
+        if not isinstance(i, str):
+            raise OptionException('Array choice elements must be strings.')
+    value = kwargs.get('value', choices[0])
+    if not isinstance(value, list):
+        if ',' in value:
+            raise OptionException('Array choices must be passed as an array, not a comma delemited string.')
```

Comment:
"`delimited`", but I agree with @nirbheek, it's probably best to be consistent and always raise this exception (which means the bit after `,` goes away anyway)